dev/breeze/src/airflow_breeze/utils/path_utils.py

```diff
@@ -97,7 +97,7 @@ def get_package_setup_metadata_hash() -> str:
     return "NOT FOUND"
 
 
-def get_sources_setup_metadata_hash(sources: Path) -> str:
+def get_pyproject_toml_hash(sources: Path) -> str:
     try:
         the_hash = hashlib.new("blake2b")
         the_hash.update((sources / "dev" / "breeze" / "pyproject.toml").read_bytes())
```

Comment:
Maybe missing context for me but why is the hash generated on a different TOML than the project? Is this for the setup of breeze itself? If it is for breeze itself we should rename the function to make clear it is the breeze hash. Otherwise you could use the beautiful new constant from above.
```suggestion
        the_hash.update((sources / "dev" / "breeze" / PYPROJECT_TOML).read_bytes())
```

Reply:
This is  because it is part of `breeze` code. It checks if `breeze` should self-upgrade. Sometimes whe you run breeze it will self-upgrade - because it will see that it's own dependencies changed (it's own `pyproject.toml`) and I have a clever mechanism to check and run it. It's described in https://github.com/apache/airflow/blob/main/dev/breeze/doc/adr/0010-use-pipx-to-install-breeze.md

I had to update that part because setup.cfg was internaly used in `breeze` to find the sources of Airflow (and now the file is gone).  However I think it will be better to extract out that change and merge it first in a separate PR - this way people who will already have `breeze` installed will old check will fail. I will separate it out  (and keep this comment to remember about it).

Reply:
Extracted to here: https://github.com/apache/airflow/pull/36537